### DISCUSSION TITLE
X-axis Twist Issue

### DISCUSSION THREAD

**Discussion by bassamanator** [bassamanator/everything-sovol-sv06#22] [Ideas]
This topic deserves a long thread, and hopefully, a simple, straightforward write-up on how to fix it.

I will try to keep this thread curated.

Use write-ups:
1. Find an explanation of the issue [here](https://github.com/Klipper3d/klipper/pull/6048).
    - Accompanying [part](https://www.printables.com/model/413616-sv06-top-z-axis-rod-fixing-blocks-parametric).
2. [This](https://www.printables.com/model/433884-sovol-sv06-x-twist-fix-top-bar-upgrade) was also helpful in visualizing the problem some more.

UPDATE:
- Klipper x-twist compensation is working well, follow discussion [here](https://github.com/bassamanator/Sovol-SV06-firmware/discussions/81).

**Comment by BeauSlim** [comment]
As requested, here's my explanation of the Z-nut-lift issue I see on my SV06:

With the hotend all the way over to the left, remove the 2 bolts holding the right POM nut and spin the right motor so the black nut moves up an inch or so. Notice that there is some play between the POM nut and the screw it rides on. Some play is is normal, as the weight of the gantry is supposed to hold the nuts to the bottom thread surface of the lead screw.

You may also see that the X gantry doesn't droop down on the right side even though nothing is holding it up. This force is what is lifting the right POM nut off of the thread it is supposed to ride. As the hotend moves to the right, the right side of the gantry slowly moves down due to the added weight, creating a curved path for the probe.

**Comment by BeauSlim** [comment]
Trying to solve this, I have installed anti-backlash POM nuts (4mm lead, https://www.aliexpress.com/item/1005002212236572.html).    I think they do help a bit but don't solve the problem completely.  

A couple of days ago I discovered that loosening the X belt to minimum tension makes a difference.  This makes sense, since the belt pulls the 2 ends of the X gantry together making it more rigid.  Loosening the belt allows the gantry to flex more.

**Reply by albertoabrao** [reply]
In regards to belt tension, [this](https://www.printables.com/model/447036-belt-tension-tester) has helped me a lot.

If left to my own devices, they always end up too tight...

**Comment by bassamanator** [comment]
Will be trying [this solution](https://github.com/bassamanator/Sovol-SV06-firmware/discussions/81) soon.

**Comment by albertoabrao** [comment]
Other than [the part I linked above ](https://www.printables.com/model/447036-belt-tension-tester)for measuring tension, I had zero issues since installing [this](https://www.printables.com/model/462039-sovol-sv06-simple-zaxis-bearing-adapter) and [this](https://www.amazon.ca/gp/product/B07JL1QYLS/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1).

The couplers linked are not "bouncy" as others I've seen, they're quite stiff, but still give way if you have any banding on the Z rods.

One thing I also do is to use [these](https://www.printables.com/model/528721-sv06sv06-plus-z-axis-leveling-tool-holder) to level the gantry, placing them on top of where the Z motors are mounted, one on each side. Scale them taller on your slicer when printing. After that, home, lift until you can fit them under the bottom rail, go from there. Many ways to skin this cat, but I personally make sure one side is rubbing just a hair, then ensure the same for the other by hand - keeping the motors powered to ensure I feel the "clicks". Do that a couple of times, and they're synchronized. Measure Z offset again - just to be sure, then ABL. So far, so good!

(yes, I am a victim of the issue where the Z leveling done by the printer leaves me with a / for an X axis...)

**Reply by albertoabrao** [reply]
Oh, one more thing: a dollar-store Square Ruler does wonders to make sure the gantries are all nice and proper. Don't go crazy, but make sure they're reasonably square!

The goal is to reduce all other mechanical issues *before* going the software route. Since I've done all of that, I haven't worried about any of it. I check Z offset and take a fresh mesh once in a while, but other than that, it just works.

**Reply by BeauSlim** [reply]
"They" generally recommend an oldham coupler for the Z-screw nut in addition to the flexible motor coupler if you are adding a top-of-Z bearing.  But, hey, all that matters is print quality so whatever works on *your* printer!  

My solution, which has been stable for quite some time now:  
I went the other way (i.e crazy) with regards to squaring the frame, by tearing the whole thing down and reassembling on a mirror with a large metal square.  I replaced the top bar with a 2020 extrusion because my plastic Sovol one was warped in 2 dimensions.  I removed the power supply and added a couple lightweight angle brackets to stop the upper frame from racking forward/back.  I added some metal brackets to stop the Z motors from sagging.   My Z screws now have zero wobble at the top.  No bearings needed.  

I am using anti-backlash POM nuts on the Z screws.  They probably aren't helping much with respect to square/level, but they definitely stop the screws from going out of sync when the printer is off.  I rarely have to relevel the gantry; I just power on and print.
